llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp

```diff
@@ -2588,6 +2588,20 @@ bool IRTranslator::translateKnownIntrinsic(const CallInst &CI, Intrinsic::ID ID,
                          getOrCreateVReg(*CI.getOperand(0)),
                          getOrCreateVReg(*CI.getOperand(1)));
     return true;
+  case Intrinsic::vector_extract: {
+    ConstantInt *Index = cast<ConstantInt>(CI.getOperand(1));
+    MIRBuilder.buildExtractSubvector(getOrCreateVReg(CI),
+                                     getOrCreateVReg(*CI.getOperand(0)),
+                                     Index->getZExtValue());
+    return true;
+  }
+  case Intrinsic::vector_insert: {
```

Comment:
What would it take to reuse the implementation of `IRTranslator::translateInsertElement` directly? (And same question for extract)

Reimplementing things here will lead us to having to fix the same issues we had in the past.
E.g.,
```
  // If it is a <1 x Ty> vector, use the scalar as it is
  // not a legal vector type in LLT.
  if (auto *FVT = dyn_cast<FixedVectorType>(U.getType());
      FVT && FVT->getNumElements() == 1)
    return translateCopy(U, *U.getOperand(1), MIRBuilder);
```

Reply:
`G_INSERT_SUBVECTOR` takes two vectors and an index and returns a vector. The second argument is inserted into the first argument. If the second vector would be a `<1 x Ty> vector`, we would still need to do a `G_INSERT_VECTOR_ELT`.

Reply:
Ah right, missed it was subvector.